ddcommon/src/lib.rs

```diff
@@ -79,48 +79,45 @@ where
     builder.build().map_err(Error::custom)
 }
 
-// TODO: we should properly handle malformed urls
-// * For windows and unix schemes:
-//     * For compatibility reasons with existing implementation this parser stores the encoded path
-//       in authority section as there is no existing standard
-//       [see](https://github.com/whatwg/url/issues/577) that covers this. We need to pick one hack
-//       or another
-// * For file scheme implementation will simply backfill missing authority section
+/// TODO: we should properly handle malformed urls
+/// * For windows and unix schemes:
+///     * For compatibility reasons with existing implementation this parser stores the encoded path
+///       in authority section as there is no existing standard [see](https://github.com/whatwg/url/issues/577)
+///       that covers this. We need to pick one hack or another
+///     * For windows, interprets everything after windows: as path
+///     * For unix, interprets everything after unix:// as path
+/// * For file scheme implementation will simply backfill missing authority section
 pub fn parse_uri(uri: &str) -> anyhow::Result<hyper::Uri> {
-    let scheme_pos = if let Some(scheme_pos) = uri.find("://") {
-        scheme_pos
+    if let Some(path) = uri.strip_prefix("unix://") {
+        encode_uri_path_in_authority("unix", path)
+    } else if let Some(path) = uri.strip_prefix("windows:") {
+        encode_uri_path_in_authority("windows", path)
```

Comment:
After checking with @gleocadie , this is actually the format of windows pipe urls, no "//".
This fucntion is only used in the telemetry lib, which is not yet used on windows so this should not be a breaking change.